### PR TITLE
Renamed workspace name from `io_bazel_rule_rust` to `rules_rust`

### PR DESCRIPTION
**Pull request by UebelAndre**
Generated by running
```bash
LANG=C sed -i '' -e 's/io_bazel_rules_rust/rules_rust/' $(find . -type f -not -path "./.git/*")
```

There's some small amount of sorting in here as well (based on the new name)

Closes https://github.com/bazelbuild/rules_rust/issues/499

Blocked by https://github.com/google/cargo-raze/pull/298 and https://github.com/google/cargo-raze/issues/304

### CODE REVIEW

**Comment by UebelAndre** [comment]
@mfarrugi with the release of [cargo-raze 0.8.0](https://github.com/google/cargo-raze/commit/4b1bf2225e79ea87639c9b814acd63b3c0c166c6) this PR is ready to go.

**Comment by UebelAndre** [comment]
@mfarrugi Just a heads up, It'd be great to get this in in a timely manner given the messaging in `cargo-raze`:
```
WARNING: The default of `[*.raze.rust_rules_workspace_name]` will soon be set to `"rules_rust"`. Please explicitly set this flag to prevent a change in behavior or upgrade your code to use the latest version of `rules_rust` and change references of `io_bazel_rules_rust` to `rules_rust` in your project.
```
I understand this is has larger implication for users but I feel confident that with the snippet above in this PR (or just a find and replace in common editors), users can easily update the their projects to allow for this switch.

**Comment by UebelAndre** [comment]
@illicitonion Hey, @mfarrugi seems to have been away for some time, would you be able to review this PR?

**Comment by acmcarther** [comment]
This should probably not be merged. This ruleset is named "io_bazel" to be consistent with the other rules defined in the bazelbuild org.

See rules go: https://github.com/bazelbuild/rules_go. Their workspace is named "io_bazel_rules_go". I don't know the history behind this naming scheme, but I do know that rules_go has "high-tier" support internally, so we should follow their lead on things.

I'm going to mirror this into the issue.

**Comment by UebelAndre** [comment]
> This should probably not be merged. This ruleset is named "io_bazel" to be consistent with the other rules defined in the bazelbuild org.
> 
> See rules go: https://github.com/bazelbuild/rules_go. Their workspace is named "io_bazel_rules_go". I don't know the history behind this naming scheme, but I do know that rules_go has "high-tier" support internally, so we should follow their lead on things.
> 
> I'm going to mirror this into the issue.

This is in direct violation of the documentation in https://docs.bazel.build/versions/master/skylark/deploying.html#workspace

> If your rules belong to the bazelbuild organization, you must use rules_<lang> (e.g. rules_mockascript)

I would assert either that documentation needs to change or this PR should be merged in a reasonable time.

**Comment by UebelAndre** [comment]
@mfarrugi re https://github.com/google/cargo-raze/pull/323#issuecomment-749433550

I'd like to talk about potential backwards compatibility shims here. I don't know of many options. To me this change is something users would have to deal with if they wanted to continue to accept changes from master. My stance is still that the changes here have a low impact on users since they can run the command in the description to update. Happy to make a shim if someone can explain what that looks like though. But I feel the sooner these changes are in the better for everyone. Otherwise things are likely to go the way of [rules_go](https://github.com/bazelbuild/rules_go/blob/master/WORKSPACE#L1) where it's acquired so many users that it becomes much more difficult to fix the name.

**Comment by mfarrugi** [comment]
The only option I know of is a bind, but I have never used it and am not
confident that it actually works.

**Comment by UebelAndre** [comment]
[bind](https://docs.bazel.build/versions/master/be/workspace.html#bind) only appears to work on targets. Not entire repositories.

**Comment by UebelAndre** [comment]
@mfarrugi Yeah, I'm not able to get this to work with `bind`. Can you think of any other options?

**Comment by acmcarther** [comment]
In the other link you referenced rules python which already made this change, so I'm not concerned about this being an issue anymore. Discussions about impact on users might still be prudent though.

**Comment by UebelAndre** [comment]
@mfarrugi I tried asking in the [slack channel](https://bazelbuild.slack.com/archives/CA31HN1T3/p1608672195323600) but also wasn't able to find great solutions for backward compatibility. I think the only thing to do would be to pull the trigger. I discussed this with @illicitonion a bit and he suggests that this be coupled a new release model (https://github.com/bazelbuild/rules_rust/issues/415) and official support for Bazel 4.0 where the min supported version follows the new LTS model introduced in that version of Bazel. Since very little changes to add 4.0 as a min version (since the rules, according to bazelisk, is already compatible). So I feel this could be merged and announcements can follow about changes to compatibility.

**Comment by UebelAndre** [comment]
I spent quite a lot of time looking into ways ways to support both workspace names. The two blockers are both related to the use of generated repositories and how they refer to `rules_rust`.

1. TODO
2. TODO

File: rust/platform/triple_mappings.bzl [hunk 1]
```diff
@@ -186,9 +186,15 @@ def triple_to_constraint_set(triple):
         list: A list of constraints (each represented by a list of strings)
     """
     if triple == "wasm32-wasi":
-        return ["@io_bazel_rules_rust//rust/platform/cpu:wasm32", "@io_bazel_rules_rust//rust/platform/os:wasi"]
+        return [
+            "@rules_rust//rust/platform/cpu:wasm32",
```
Comment (UebelAndre) [inline rust/platform/triple_mappings.bzl:190]: These strings get rendered into the BUILD files for the generated toolchains. There needs to be a way to detect the current `rules_rust` workspace name and render that into there. There may also be some potential for `repo_mapping` but that may also require the ability to dynamically detect the workspace name or force users to explicitly pass this data which to me seems undesirable.

File: proto/raze/remote/BUILD.base64-0.9.3.bazel [hunk 2]
```diff
@@ -7,7 +7,7 @@ DO NOT EDIT! Replaced on runs of cargo-raze
 
 # buildifier: disable=load
 load(
-    "@io_bazel_rules_rust//rust:rust.bzl",
+    "@rules_rust//rust:rust.bzl",
```
Comment (UebelAndre) [inline proto/raze/remote/BUILD.base64-0.9.3.bazel:10]: This BUILD file (as with all the others generated via `cargo-raze` treats `rules_rust` as an external repository. The `rust_rules_workspace_name` setting can be used to control the name but in order to really support multiple names there would need to be multiple versions of these files that have the new name. Again, the blocker here is the inability to detect the workspace name dynamically.

**Comment by UebelAndre** [comment]
@mfarrugi @damienmg @illicitonion This PR is ready to go. I'm not sure what else should be done for this. @illicitonion talked about doing the rename at the same time as the 4.0 migration (which I think is a no-op since the rules are already compatible). Maybe https://github.com/bazelbuild/rules_rust/issues/445 can be done at the same time? And that would conclude "breaking" changes for the near future?

**Comment by damienmg** [comment]
I can take a look at what is needed for #445 next week but I don't think it should block this PR.

**Comment by UebelAndre** [comment]
I'm happy to have this merged 😃  is there anything else I should do here?

**Comment by UebelAndre** [comment]
@damienmg hey, just pinging this again 😅

**Review by damienmg — APPROVED** [review]

**Comment by damienmg** [comment]
Sorry I cannot address #445 at the moment, I am fighting fires at work and in my personal life :(

Related to #445, I received a bug that we have `whitelist` somewhere in our codebase, suggesting we replaced it with allowlist (definitely a good idea IMO).

**Comment by UebelAndre** [comment]
Thanks @damienmg! I've really enjoyed having you as a reviewer and would love to keep working with you. I hope everything bounces back for you and am wishing you the best 😄

**Pull request merged**